File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size();
        // sort(nums.begin(), nums.end());
        // for(int i = 0; i< n-1; i++){
        //     if(nums[i] == nums[i+1]){
        //         return true;
        //     }
        // }
        // return false;

        // map<int, int> mp;

        // for(auto it: nums){
        //     mp[it]++;
        // }
        // for(auto it: mp){
        //     if(it.second > 1){
        //         return true;
        //     }
        // }
        // return false;
        
        set<int> s;

        for(int i = 0; i< n; i++){
            s.insert(nums[i]);
        }
        if(s.size() < n){
            return true;
        }
        else{
            return false;
        }

    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    
    let orderSet = new Set();

    // return orderSet.size !== nums.length;

    for(let num of nums){
        if(orderSet.has(num)){
            return true;
        }
        orderSet.add(num)
    }
    return false;

};